### PR TITLE
added option to disable auto refresh

### LINKED ISSUE #208: [Feature] Configurable automatic refreshing
When I stage a hunk on `DiffviewOpen` the diff is refreshing and its annoying, you have any idea why it happens and how can I stop it?

### PR DESCRIPTION
**Pull request by ofirgall**
Resolves #208.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -154,6 +154,7 @@ require("diffview").setup({
   enhanced_diff_hl = false, -- See ':h diffview-config-enhanced_diff_hl'
   git_cmd = { "git" },      -- The git executable followed by default args.
   use_icons = true,         -- Requires nvim-web-devicons
+  auto_refresh = true,      -- Auto refresh the status of the current file
```
Comment (sindrets) [inline README.md:157]: Let's call it `watch_index`. And I think we should make the description a little bit more accurate:

```suggestion
  watch_index = true,       -- Update views and index buffers when the git index changes.
```

File: lua/diffview/scene/views/diff/diff_view.lua [hunk 2]
```diff
@@ -124,15 +125,17 @@ function DiffView:post_open()
 
   self.watcher = vim.loop.new_fs_poll()
   ---@diagnostic disable-next-line: unused-local
```
Comment (sindrets) [inline lua/diffview/scene/views/diff/diff_view.lua:127]: Please move both of these lines inside the if clause you created. No need to create and hold on to a file descriptor we aren't using.

**Review by sindrets — CHANGES_REQUESTED** [review]
Thanks for the PR! I left some comments.

**Comment by ofirgall** [comment]
Made the changes

**Review by sindrets — APPROVED** [review]
LGTM, thanks!

**Pull request merged**